### ISSUE TITLE
Make it clear which annotations can and can not be preserved with save tool

### ISSUE THREAD

**Issue by DuCorey** [bokeh/bokeh#6591]
When saving a plot figure using the save tool items rendered with css will not appear on the saved image. I've found this behavior consistent for Labels and Titles. 

Exploring this issue further using the new export_png and export_svg. I've found that using export_png **will** generate the figure with all the css.

However, using the export_svg **will not** generate the figure with css. Also while using the save tool with the `output_backend = svg` , the figure once saved in .svg **will not** contain any rendered css elements.

#### Complete, minimal, self-contained example code that reproduces the issue
For a quicker example you can simply try to save the image on http://bokeh.pydata.org/en/latest/docs/user_guide/annotations.html#labels and you 
will notice that  "Collected by Luke C. 2016-04-01" is not saved to the image.
![bokeh_plot 36](https://user-images.githubusercontent.com/27960786/27970766-f27b078a-631d-11e7-97a3-db3e7240d79e.png)

Here's some example code that I used.
```
from bokeh.plotting import figure, show, output_file
from bokeh.models import Label
# from bokeh.io import export_png, export_svgs

x = [1, 2, 3, 4]
y = [1, 2, 3, 4]

p = figure()
p.title.text = "This is a css title"
p.title.render_mode = 'css'

p.circle(x=x, y=y)

label = Label(x=2.5, y=2.5, text='This is a css label', render_mode='css')
p.add_layout(label)

# export_png(p, filename="plot.png")
output_file("css_bokeh.html", title="css_bokeh bugs")

# p.output_backend = "svg"
# export_svgs(p, filename="plot.svg")

show(p)
```

#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
Browser: Chrome 59.0.3071.115
OS: Windows 7

```
- selenium=3.4.3=py35_0
- bkcharts=0.2=py35_0
- bokeh=0.12.6=py35_0
- bzip2=1.0.6=vc14_3
- freetype=2.5.5=vc14_2
- jinja2=2.9.6=py35_0
- jpeg=9b=vc14_0
- libpng=1.6.27=vc14_0
- libtiff=4.0.6=vc14_3
- markupsafe=0.23=py35_2
- mkl=2017.0.1=0
- numpy=1.13.0=py35_0
- olefile=0.44=py35_0
- phantomjs=2.1.1=0
- pillow=4.2.0=py35_0
- pip=9.0.1=py35_1
- python=3.5.3=3
- python-dateutil=2.6.0=py35_0
- pyyaml=3.12=py35_0
- requests=2.14.2=py35_0
- setuptools=27.2.0=py35_1
- six=1.10.0=py35_0
- tornado=4.5.1=py35_0
- vs2015_runtime=14.0.25123=0
- wheel=0.29.0=py35_0
- zlib=1.2.8=vc14_3
```

**Comment by bryevdv** [comment]
Hi, the current situation os probably the only possible situation. 

* the save tool exports the canvas pixels and nothing else. This is probably all it will ever be capable of

* the png export uses selenium and either phantoms or chrome headless to get a "screenshot"  so it can capture things that are not on the canvas (like labels that are dome elements)

* the svg export uses a canvas shim to change HTML canvas commands into SVG. it only knows about canvases. 

The reality is that there are multiple (partially) overlapping solutions, and technical limitations mean that will probably always be the case

**Comment by bryevdv** [comment]
I should add, I'd *love* to be proven wrong, but after thinking and working on various export ideas over several years now, I am not optimistic. Certainly for now, if you need DOM elements in the exported image image, PNG is your only option. We should probably add more information about the various cases and limitations to the documentation though.

**Comment by DuCorey** [comment]
@bryevdv I've got a gist of it working on my end. Here's the [link](https://gist.github.com/DuCorey/abb71f30e2c89e5178259e5c5408b505). It uses an external library ([dom-to-image](https://github.com/tsayen/dom-to-image)) to combine multiple dom sub elements into a single canvas. Running the function on the bk-canvas-wrapper allows it to include both the original canvas and the bk-canvas-overlays. 

Since it's an external library, I would like to know your opinion on if this way of doing it is good idea and if it is possible to include it in the Bokeh code. I'm still new to open source development in general so please forgive my ignorance. Let me know what you think on how to proceed from here.

**Comment by bryevdv** [comment]
@DuCorey no apologies, this seems like an interesting angle that we were not aware of (definitely was not around last time I did a survey) I think we will have to discuss and consider how it might be useful. ping @canavandl any thoughts?

**Comment by mattpap** [comment]
I will consider this resolved when PR #11621 is merged. This PR introduces clear separation between canvas and HTML rendered annotations, meaning annotations that can and can not be preserved when using save tool or its variants.
